src/python/aqueduct_executor/operators/function_executor/execute.py

```diff
@@ -216,21 +218,49 @@ def run(spec: FunctionSpec) -> None:
     """
     Executes a function operator.
     """
-    print("Started %s job: %s" % (spec.type, spec.name))
-
     exec_state = ExecutionState(user_logs=Logs())
     storage = parse_storage(spec.storage_config)
     try:
         validate_spec(spec)
 
         # Read the input data from intermediate storage.
+        print_with_color(
+            "Reading inputs for %s job: %s" % (spec.type.value, spec.name),
+            color=PrintColorType.GREEN,
+        )
+        begin = time.time()
+
         inputs, _, serialization_types = utils.read_artifacts(
             storage, spec.input_content_paths, spec.input_metadata_paths
         )
 
+        end = time.time()
+        performance = {
+            "job": spec.name,
+            "type": spec.type,
+            "step": "Reading Inputs",
+            "latency(s)": (end - begin),
+        }
+        print_with_color(json.dumps(performance, indent=4))
+
         derived_from_bson = SerializationType.BSON_TABLE in serialization_types
-        print("Invoking the function...")
+
+        print_with_color(
+            "Running function for %s job: %s" % (spec.type.value, spec.name),
+            color=PrintColorType.GREEN,
+        )
+        begin = time.time()
         results, system_metadata = _execute_function(spec, inputs, exec_state)
+        end = time.time()
```

Comment:
The repetitiveness of this pattern makes me think this latency logging should be a decorator. Do you think that would be a possible change we could make here? It would really clean up the code and prevent bugs.

Reply:
I would imagine something simple like this would suffice: https://dev.to/kcdchennai/python-decorator-to-measure-execution-time-54hk

Of course, you would pass in the variables needed to print everything correctly.